.eslintrc.json

```diff
@@ -18,6 +18,15 @@
 			"rules": {
 				"@typescript-eslint/array-type": "error",
 				"@typescript-eslint/ban-ts-comment": 0,
+				"@typescript-eslint/ban-types": [
+					"error",
+					{
+						"types": {
+							"{}": false
```

Comment:
this rule is dumb so lets disable it. using `{}` in type intersections is actually more than fine.

Reply:
It kinda makes sense to enforce `Record<string, unknown>` in places where you'd typically use just `{}` :shrug:

Reply:
those are very different things, i am allowing this to be able to use `{}` in intersections

Reply:
I get that that's the intention and it makes sense, but is it what that rule does? I'm asking because I don't know :smile:

Reply:
I'm super unsure why you'd need to use `NonNullable` in the places where you added it then :slightly_smiling_face: but I don't want to stall the PR, so no need to dwell on this.